### DISCUSSION TITLE
Support of gcc 4.x - will it ever be dropped?

### DISCUSSION THREAD

**Discussion by sjanel** [nlohmann/json#3830] [General]
Hello,

I am wondering why json still (in 2022) needs to support a compiler that is legacy, especially considering that there is the term "Modern" is the headline of the library.

As a reminder, gcc < v5 does *not* even support C++11. There are several traits missing (for instance `std::is_trivially_copyable` amongst others) and the compiler is even buggy for some code, that prevents optimizations.

In a PR of mine I wanted to give an interesting optimization of the `json::iterator` and the code had not been approved just because it could not compile with gcc 4 only, which is really a shame for a library claming to be for modern C++.

**Comment by nlohmann** [comment]
Do you mean #3452?

**Reply by sjanel** [reply]
Yes indeed!

**Reply by nlohmann** [reply]
I don't see GCC 4 as blocker there. Can you resolve the conflicts and address the open issues?

**Reply by sjanel** [reply]
I can, but yes GCC 4.8 is a blocker. I provided workarounds based on conditional compiling to make the code compile with gcc 4.8 but it's ugly and there is no other way. Also, the optimization relies on trait std::is_trivially_copyable which is not present in GCC v < 5 so I had to consider it as false for all types.

**Reply by sjanel** [reply]
> I don't see GCC 4 as blocker there. Can you resolve the conflicts and address the open issues?

I answered to all of the requests but got no answers. Unfortunately the proposed solutions for gcc 4.8 do not work, and I posted godbolt's links to illustrate my points.

**Comment by Rohan427** [comment]
I am waaay late to the party, but only because I am looking for a JSON library for C/C++ for an application that this question is pertinent to.

gcc 4.x is relevant (to my dismay - but it takes time to update some things that already "just work") for some government applications, such as the one I am working on (no, I can't say what it is). I can say the version is gcc 4.4.7 (yes, a broken compiler, but it is what it is).

**Reply by nlohmann** [reply]
No worries - I am not planning to drop GCC 4.x support - at least GCC 4.8. Does anything work at all with 4.4.7?
